main.py

```diff
@@ -0,0 +1,206 @@
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here.
+            return
+        return func(self, opt)
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+border = colored("*" * 20, 'cyan').center(80)
+
+
+def introduction():
+    print (border)
+    print ("WELCOME TO AMITY SPACE ALLOCATION!".center(70))
+    print(__doc__)
+    print (border)
+
+
+class AmityApplication(cmd.Cmd):
+    cprint(figlet_format('AMITY', font='banner3-D'), 'cyan', attrs=['bold'])
+    prompt = "Amity -->"
+    amity = Amity()
+
+    @docopt_cmd
+    def do_create(self, arg):
+        '''Usage: create room <room_type> <room_name> ...'''
+        try:
+            name = arg['<room_name>']
+            room_type = arg['<room_type>']
+            print(self.amity.create_room(room_type+' '+name[0]))
+        except Exception as e:
+            print(e)
+
+    @docopt_cmd
+    def do_check(self, arg):
+        '''Usage: check (room_exists|room_availability|person_exist) <name>'''
+        try:
+            name = arg['<name>']
+            if arg['room_exists']:
+                print(self.amity.room_exists(name))
+            elif arg['room_availability']:
+                print(self.amity.check_room_availability(name))
+            elif arg['person_exist']:
+                print(self.amity.person_exists(name))
+        except Exception as e:
+            print(e)
+
+    @docopt_cmd
+    def do_add(self, arg):
+        '''Usage: add person <first_name> <last_name> <FELLOW|STAFF>
+        [wants_accomodation]'''
+        try:
+            name = arg['<first_name>'] + ' ' + arg['<last_name>']
+            if arg['<FELLOW']:
+                typ = arg['<FELLOW']
+            if arg['STAFF>']:
+                typ = arg['STAFF>']
+            person = name + ' ' + typ
+            if arg['wants_accomodation']:
+                acc = arg['wants_accomodation']
+                print(self.amity.add_person(person+' '+acc))
+            else:
+                print(self.amity.add_person(person))
+        except Exception as e:
+            print(e)
+
+    @docopt_cmd
+    def do_allocate(self, arg):
+        '''Usage: allocate space <first_name> <last_name> <person_type>
+        [wants_accomodation]'''
+        try:
+            person = arg['<first_name>'] + ' ' + arg['<last_name>']
+            person = {
+                'name': person,
+                'type': arg['<person_type>']
+            }
+            if arg['want_accomodation']:
+                print(self.amity.allocate_space(person, True))
+            else:
+                print(self.amity.allocate_space(person))
+        except Exception as e:
+            print(e)
+
+    @docopt_cmd
+    def do_reallocate(self, arg):
+        '''Usage: reallocate person <first_name> <last_name> <new_room_name>'''
+        try:
+            person = arg['<first_name>'] + ' ' + arg['<last_name>']
```

Comment:
use more descriptive variable name i.e `person_name` here to clearly communicate the meaning of the variables